packages/render-fidelity-tools/src/common.ts

```diff
@@ -253,6 +279,13 @@ export class ImageComparator {
       }
     }
 
+    const imagePixelCount = width * height;
+    if (whitePixelCount === imagePixelCount ||
+        blackPixelCount === imagePixelCount ||
```

Comment:
This is subtly different than the logic I was trying to describe. This will fail only if the image is all white or all black or all transparent. We want it to fail none of the pixels are non- white, black, transparent. So you only need to keep count of one category, instead of the three separately.